io.github.sxyazi.yazi.yml

```diff
@@ -0,0 +1,263 @@
+      # Workaround for Go modules generated by github.com/dennwc/flatpak-go-mod
+      - modules/host-spawn/sources.yml
+  # For previewing the content of a zip file
+  - name: p7zip
+    make-args:
+      - 7z
+    no-autogen: true
+    make-install-args:
+      - DEST_HOME=/app
+    sources:
+      - type: archive
+        url: https://github.com/p7zip-project/p7zip/archive/refs/tags/v17.05.tar.gz
+        sha256: d2788f892571058c08d27095c22154579dfefb807ebe357d145ab2ddddefb1a6
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/p7zip-project/p7zip/releases/latest
+          url-query: '"https://github.com/p7zip-project/p7zip/archive/refs/tags/\($version).tar.gz"'
+          version-query: .tag_name
+  # For pdf previews
+  - name: poppler
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DBUILD_GTK_TESTS=OFF
+      - -DBUILD_QT5_TESTS=OFF
+      - -DBUILD_QT6_TESTS=OFF
+      - -DBUILD_CPP_TESTS=OFF
+      - -DENABLE_CPP=ON
+      - -DENABLE_GLIB=OFF
+      - -DENABLE_QT5=OFF
+      - -DENABLE_QT6=OFF
+      - -DENABLE_BOOST=OFF
+    sources:
+      - type: archive
+        url: https://poppler.freedesktop.org/poppler-24.12.0.tar.xz
+        sha256: 1cf374c3146f3f685d9257701bf0c2866c61d6c202c14d1f5c01a1f3a089028a
+        x-checker-data:
+          type: anitya
+          project-id: 3686
+          stable-only: true
+          url-template: https://poppler.freedesktop.org/poppler-$version.tar.xz
+  # Main module
+  - name: yazi
+    buildsystem: simple
+    build-options:
+      env:
+        - CARGO_HOME=/run/build/yazi/cargo
+    build-commands:
+      - cargo --offline fetch --manifest-path Cargo.toml
+      - cargo --offline build --release
+      - mkdir -p /app/lib/ffmpeg
+      - install -Dm00755 target/release/yazi ${FLATPAK_DEST}/bin/yazi
+      # `ya` is a package manager for themes/plugins
+      - install -Dm00755 target/release/ya ${FLATPAK_DEST}/bin/ya
+      # Upstream icon is 1254x1260. Shrink to 512x512 to conform to icon standard. 
+      - ffmpeg -hide_banner -i assets/logo.png -vf scale=512:512 logo.png
+      - install -Dm00644 logo.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      # The icon specified in the upstream dot desktop file does not play nicely with Flatpak.
+      - desktop-file-edit assets/yazi.desktop --set-key=Icon --set-value="${FLATPAK_ID}"
+      - install -Dm00644 assets/yazi.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm00644 ${FLATPAK_ID}.appdata.xml -t ${FLATPAK_DEST}/share/metainfo
+    sources:
+      - type: git
+        url: https://github.com/sxyazi/yazi.git
+        tag: v0.4.2
+        commit: 5cfcab305ef0a02771b3bd651ed343e436fc7f7e
+        x-checker-data:
+          type: anitya
+          project-id: 370571
+          stable-only: true
+          tag-template: v$version
+      - type: dir
+        path: .
```

Comment:
don't use dir sources, these cannot be cached.

Reply:
Done. Not sure why I had them. Probably a blanket copy-paste from another repo.